From aa7069ac138056af1e16496b47e2545def936370 Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:16:03 -0400
Subject: [fs] cifs: Allow LANMAN auth method for servers supporting unencapsulated authentication methods

Message-id: <1406549822-10376-11-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86748
O-Subject: [RHEL7.1 PATCH 10/69] [bz1082049]cifs: Allow LANMAN auth method for servers supporting unencapsulated authentication methods
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

This allows users to use LANMAN authentication on servers which support
unencapsulated authentication.

The patch fixes a regression where users using plaintext authentication
were no longer able to do so because of changed bought in by patch
3f618223dc0bdcbc8d510350e78ee2195ff93768

https://bugzilla.redhat.com/show_bug.cgi?id=1011621

Reported-by: Panos Kavalagios <Panagiotis.Kavalagios@eurodyn.com>
Reviewed-by: Jeff Layton <jlayton@redhat.com>
Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
Signed-off-by: Steve French <smfrench@gmail.com>
(cherry picked from commit dde2356c8466298bd77fa699e0ea296372eed47b)
Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/sess.c b/fs/cifs/sess.c
index 352358d..e87387d 100644
--- a/fs/cifs/sess.c
+++ b/fs/cifs/sess.c
@@ -500,9 +500,9 @@ select_sectype(struct TCP_Server_Info *server, enum securityEnum requested)
     return NTLMv2;
    if (global_secflags & CIFSSEC_MAY_NTLM)
     return NTLM;
-   /* Fallthrough */
   default:
-   return Unspecified;
+   /* Fallthrough to attempt LANMAN authentication next */
+   break;
   }
  case CIFS_NEGFLAVOR_LANMAN:
   switch (requested) {
-- 
1.7.1